msal/tests/Test.MSAL.NET.Unit.net45/CacheV2Tests/FileIOTests.cs

```diff
@@ -0,0 +1,779 @@
+﻿// ------------------------------------------------------------------------------
+// 
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+// 
+// This code is licensed under the MIT License.
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Test.MSAL.NET.Common.Core.Helpers;
+#if NETFRAMEWORK
+using Microsoft.Identity.Core.Platforms.net45.CacheV2;
+#endif
+
+namespace Test.MSAL.NET.Unit.net45.CacheV2Tests
+{
+#if NETFRAMEWORK
+    [TestClass]
+    public class FileIOTests
+    {
+        private const string FileName = "FileIOTests.bin";
+        private const string TestFolderBase = "FileIOTestsFolder";
+        private const string TestFolder = "FileIOTestsFolder\\MoreFileIOTestsFolder";
+        private byte[] _data;
+        private WindowsFileSystemCacheKeyStorage _io;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            TestCleanup();
+            _io = new WindowsFileSystemCacheKeyStorage(Path.Combine(AssemblyUtils.GetExecutingAssemblyDirectory(), TestFolder));
+            _data = RandomDataUtils.GetRandomData(1024);
+        }
+
+        [TestCleanup]
+        public void TestCleanup()
+        {
+            if (Directory.Exists(TestFolderBase))
+            {
+                Directory.Delete(TestFolderBase, true);
+            }
+
+            if (File.Exists(TestFolderBase))
+            {
+                File.Delete(TestFolderBase);
+            }
```

Comment:
Seems like Directory.Delete(,true) should be able to able to handle deleting both files and sub folders?

Reply:
Directory.Delete doesn't work if you pass it a filename.  And sometimes we're deleting a file (we have a test that creates a file named what the folder should be and ensure we fail properly) and other times we're deleting a folder (the normal case).